server/base/src/main/java/org/apache/accumulo/server/gc/AllVolumesDirectory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.server.gc;
+
+import static org.apache.accumulo.server.gc.GcVolumeUtil.ALL_VOLUMES_PREFIX;
+
+import java.util.Objects;
+
+import org.apache.accumulo.core.Constants;
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.gc.ReferenceDirectory;
+import org.apache.accumulo.core.metadata.schema.MetadataSchema;
+import org.apache.hadoop.fs.Path;
+
+/**
+ * A specially encoded GC Reference to a directory with the {@link GcVolumeUtil#ALL_VOLUMES_PREFIX}
+ */
+public class AllVolumesDirectory extends ReferenceDirectory {
```

Comment:
I am trying to figure out why this extends reference.  Does not seem it is used as a reference in the GC algorithm.  Seems that `GCRun.getReferences()` will never return this type.  Conceptually I think the ALL_VOLUMES_PREFIX type is a special type of GC candidate, but not a reference.  I am still poking around in the code trying to understand the larger context, will circle back to it in the morning.

Reply:
Check out [df7f4f9](https://github.com/apache/accumulo/pull/2767/commits/df7f4f977f400c8eb5717d14bfb69587545d4008). I was able to make metadataEntry final by having AllVolumesDirectory extend ReferenceFile.

Reply:
> Thinking about this I was wondering if it would make sense to have a GcCandidate type in addition to the Reference type.

For sure. I had some ideas about a strong type for candidates and a backing Map class to replace the Map we pass around. But I was trying to keep changes to  to a minimum to be less disruptive so close to a release candidate.

> For this PR we may want to document that the reference type is also used for GC candidates.

I can add some comments.